core/src/replay_stage.rs

```diff
@@ -2894,9 +2894,7 @@ impl ReplayStage {
                         Tower::is_direct_vote_state_update_enabled(bank),
                         bank.get_vote_account(my_vote_pubkey),
                     ) {
-                        if let Some(mut bank_vote_state) =
-                            vote_account.vote_state().as_ref().ok().cloned()
-                        {
+                        if let Ok(mut bank_vote_state) = vote_account.vote_state().cloned() {
```

Comment:
doesn't this clone the `Error` and then through it away?

Reply:
i don't [_believe_ so](https://doc.rust-lang.org/std/result/enum.Result.html#method.cloned)

> Maps a Result<&T, E> to a Result<T, E> by cloning the contents of the Ok part.